### ISSUE TITLE
Travis build fails because of Licensee update

### ISSUE THREAD

**Issue by pchaigno** [github/linguist#3225]
The Travis build is currently failing on master and in all new pull requests.

I believe this has something to do with [the last Licensee release](https://github.com/benbalter/licensee/releases/tag/v8.4.0). In particular, I would guess that this is because the required confidence threshold is now 95% instead of 90%. I haven't had time to debug yet though.

**Comment by pchaigno** [comment]
I was able to confirm that the issue comes from licensee (it works when I lock the licensee version to `v4.3.1`). Here are the results of licensee `v4.3.1` against the four grammars that are now unrecognized:
```
/go-tmbundle$ licensee
License file: LICENSE
Attribution: Copyright (c) 2009, Jim Dovey
License: BSD 3-clause "New" or "Revised" License
Confidence: 94.78%
Method: Licensee::Matchers::Dice
```
```
/jflex.tmbundle$ licensee
License file: LICENSE.md
Attribution: Copyright (c) 2015 Gerwin Klein and contributors.
License: BSD 2-clause "Simplified" License
Confidence: 94.88%
Method: Licensee::Matchers::Dice
```
```
/factor$ licensee
License file: license.txt
License: BSD 2-clause "Simplified" License
Confidence: 94.23%
Method: Licensee::Matchers::Dice
```
```
/sublimeprolog$ licensee
License file: LICENSE
License: Mozilla Public License 2.0
Confidence: 100.00%
Method: Licensee::Matchers::Exact
```
The three first grammars are not recognized with the last version because of the new 95% threshold. I've opened an issue for the last one: benbalter/licensee#115.

I'm not sure what we should do for the three first grammars...
@arfon What do you think? Should be lower the threshold?
@benbalter Any particular reason for raising the threshold?

**Comment by benbalter** [comment]
> Any particular reason for raising the threshold?

Licensee was having trouble detecting when both the MIT and GPL licenses were in the same file (https://github.com/benbalter/licensee/pull/112).

Can you provide me with links to the failing grammars' licenses? I'd love to dig into why they're failing.

**Comment by Alhadis** [comment]
Well, [here's one example](https://travis-ci.org/github/linguist/jobs/160276457#L1217):

```
  2) Failure:
TestGrammars#test_submodules_have_approved_licenses [/home/travis/build/github/linguist/test/test_grammars.rb:93]:
The following submodules have unapproved licenses:
* vendor/grammars/go-tmbundle: 
* vendor/grammars/factor: 
* vendor/grammars/jflex.tmbundle: 
* vendor/grammars/sublimeprolog: gpl-2.0
The license must be added to the LICENSE_WHITELIST in /test/test_grammars.rb once approved..
--- expected
+++ actual
@@ -1 +1 @@
-[]
+["vendor/grammars/go-tmbundle: ", "vendor/grammars/factor: ", "vendor/grammars/jflex.tmbundle: ", "vendor/grammars/sublimeprolog: gpl-2.0"]
```

It thinks [sublimeprolog](https://github.com/alnkpa/sublimeprolog) is released under the GPL 2.0 license, whereas it's [actually the **M**PL 2.0](https://github.com/alnkpa/sublimeprolog/blob/master/LICENSE).

**Comment by pchaigno** [comment]
> Can you provide me with links to the failing grammars' licenses? I'd love to dig into why they're failing.

Sure:
- [jflex-de/jflex.tmbundle](https://github.com/jflex-de/jflex.tmbundle/tree/12a7a88b7a34bb9490b6077ad743332acf66a456)
- [slavapestov/factor](https://github.com/slavapestov/factor/tree/d99c9e16632d2eb1e5a8862182529d3dcc6b7c56)
- [AlanQuatermain/go-tmbundle](https://github.com/AlanQuatermain/go-tmbundle/tree/0bdec37eded22e8416d046fe28e50d2c5c666798) (This one has a footnote on the license.)
- [alnkpa/sublimeprolog](https://github.com/alnkpa/sublimeprolog/tree/c71d8a1e0f162165cd93d0e41acb7a8984d0b30f)

**Comment by arfon** [comment]
Perhaps we should whitelist these grammars for now? e.g. with the `curated: true` flag we've been using [elsewhere](https://github.com/github/linguist/blob/00efd6a463d198ae5b3c0482fb66e9f7cb71e7cd/vendor/licenses/grammar/eiffel.tmbundle.txt#L5).

**Comment by benbalter** [comment]
https://github.com/benbalter/licensee/pull/117 should fix `alnkpa/sublimeprolog`. Looking into the other two.

**Comment by benbalter** [comment]
> (This one has a footnote on the license.)

This is the expected behavior. There's no way for the computer to know that that doesn't say "And you must give me one free puppy every day".

**Comment by benbalter** [comment]
> jflex-de/jflex.tmbundle

The one also has a line added. They substituted the title of the project for the license title (which again, the computer doesn't know). You can lower the confidence threshold if you'd like, but then you may get false positives, rather than false negatives as you are here, especially around concatenated licenses.

**Comment by pchaigno** [comment]
Thanks for the help @benbalter!

@arfon I guess #3236 fixes this?

**Comment by arfon** [comment]
> @arfon I guess #3236 fixes this?

Yeah, kind of :-\ Not entirely happy with that PR.

**Comment by pchaigno** [comment]
I think this has been addressed with #3251.
